scripts/template_upload_gce.py

```diff
@@ -104,51 +131,98 @@ def check_template_name(name):
     return name
 
 
-def upload_template(project, zone, service_account, image_url,
-        template_name, bucket_name, provider):
+def upload_template(provider,
+                    template_name,
+                    stream,
+                    file_name,
+                    file_path,
+                    ssh_client,
+                    bucket_name=None):
+    bucket = bucket_name or cfme_data['template_upload']['template_upload_gce']['bucket_name']
     try:
-        if not bucket_name:
-            bucket_name = cfme_data['template_upload']['template_upload_gce']['bucket_name']
-        print("GCE:{} Starting image downloading {} ...".format(provider, image_url))
-        file_name, file_path = download_image_file(image_url)
-        print("GCE:{} Image downloaded {} ...".format(provider, file_path))
-
-        print("GCE:{} Creating bucket and uploading file {}...".format(provider, bucket_name))
-        gcloud = GoogleCloudSystem(project=project, zone=zone, service_account=service_account)
-        gcloud.create_bucket(bucket_name)
-        blob_name = gcloud.get_file_from_bucket(bucket_name, file_name)
-        if not blob_name:
-            gcloud.upload_file_to_bucket(bucket_name, file_path)
-            blob_name = gcloud.get_file_from_bucket(bucket_name, file_name)
-        print("GCE:{} File uploading done ...".format(provider))
-
-        print("GCE:{} Creating template/image {}...".format(provider, template_name))
+        # IMAGE CHECK
+        log_detail('Checking if template {} present...'.format(template_name), provider)
+        result = ssh_client.run_command('gcloud compute images list {}'.format(template_name))
```

Comment:
talked offline, this is the only one that returns 0 even when it doesn't find anything, so I'm forced to rely just on stdout.
[root@[host] ~]# gcloud compute images list cfme-5-6-0-11-junk
Listed 0 items.
[root@[host] ~]# echo $?
0